AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -4295,40 +4297,39 @@ function Update-AzStorageAccountAuthForAES256 {
             -ResourceGroupName $ResourceGroupName -StorageAccountName $StorageAccountName -ErrorAction Stop
         $domain = $activeDirectoryProperties.DomainName
 
-        if ($adObject.ObjectClass -ine "computer") {
-            $message = "Removing object '$($adObject.DistinguishedName)' of type '$adObject.ObjectClass' from domain '$domain'." `
-                + " AES256 is only supported for computer objects."
-            Write-Verbose -Message $message
+        switch($adObject.ObjectClass) {
+            "user" {
+                Write-Verbose -Message "Set AD user object '$($adObject.DistinguishedName)' to use AES256 for Kerberos authentication"
+                $userPrincipalName = $adObject.UserPrincipalName
 
-            Remove-ADObject -Identity $adObject.DistinguishedName -Server $domain -Confirm:$false -ErrorAction Stop
+                if ([string]::IsNullOrEmpty($userPrincipalName)) {

```

Comment:
Can we add a check to prompt a message and stop if upn exists but is not the expected value? We probably can ask the customer to fix the upn or provide a cmdlet parameter to indicate we can override it.

Reply:
updated